components/Datepicker/src/index.tsx

```diff
@@ -0,0 +1,400 @@
+import clsx from 'clsx';
+import {
+  format,
+  Locale,
+  getWeeksInMonth,
+  formatISO,
+  setDay,
+  addWeeks,
+  subMonths,
+  addMonths,
+  isSameMonth,
+  setDate,
+  isEqual,
+  addDays,
+  subDays,
+  startOfWeek,
+  endOfWeek,
+  subYears,
+  addYears,
+  subWeeks,
+  getWeekOfMonth,
+  getDay,
+  getDate,
+  setMonth,
+  getMonth,
+  setYear,
+  getYear,
+} from 'date-fns';
+import { nl } from './locale';
+
+import { CalendarIcon, AlertTriangleIcon, ChevronRightIcon, ChevronLeftIcon } from '@gemeente-denhaag/icons';
+
+import './datepicker.css';
+
+export interface DatepickerProps extends InputHTMLAttributes<HTMLInputElement> {
+  /**
+   * The ref for the outer HTML div.
+   */
+  ref?: RefObject<HTMLDivElement>;
+  /**
+   * The ref for the actual input HTML element.
+   */
+  inputRef?: RefObject<HTMLInputElement>;
+  /**
+   * Whether the input field is in a error state.
+   */
+  error?: boolean;
+  /**
+   * Whether the input field is disabled.
+   */
+  disabled?: boolean;
+  /**
+   * Locale (from @gemeente-denhaag/datepicker/locale or date-fns/locale) (default: nl)
+   */
+  locale?: Locale;
+  /**
+   * Starting date for calendar (default: today)
+   */
+  startDate?: Date;
+  /**
+   * aria-label for the button to open the calendar.
+   */
+  openCalendarLabel?: string;
+  /**
+   * aria-label for previous month button in the calendar.
+   */
+  previousMonthLabel?: string;
+  /**
+   * aria-label for previous month button in the calendar.
+   */
+  nextMonthLabel?: string;
+}
+
+interface DatepickerState {
+  current: Date;
+  selected?: Date;
+  opened: boolean;
+}
+
+/**
+ * The DatePicker component allows the user to select a date.
+ */
+export const Datepicker: React.FC<DatepickerProps> = ({
+  ref = React.useRef<HTMLDivElement>(null),
+  inputRef = React.useRef<HTMLInputElement>(null),
+  startDate = new Date(),
+  locale = nl,
+  openCalendarLabel = 'Kies een datum',
+  previousMonthLabel = 'Vorige maand',
+  nextMonthLabel = 'Volgende maand',
+  ...props
+}: DatepickerProps) => {
+  const rootStyles = clsx(
+    'denhaag-datepicker',
+    {
+      'denhaag-datepicker--error': props.error,
+      'denhaag-datepicker--disabled': props.disabled,
+    },
+    props.className,
+  );
+
+  const [state, setState] = React.useState<DatepickerState>({
+    current: startDate,
+    selected: undefined,
+    opened: false,
+  });
+
+  const currentDate = new Date();
+  const today = setYear(
+    setMonth(setDate(state.current, getDate(currentDate)), getMonth(currentDate)),
+    getYear(currentDate),
+  );
+
+  const openButtonRef = React.useRef<HTMLButtonElement>(null);
+  const backButtonRef = React.useRef<HTMLButtonElement>(null);
+  const currentButtonRef = React.useRef<HTMLButtonElement>(null);
+
+  const outsideClickListener = (e: Event) => {
+    if (!ref?.current?.contains(e.target as Node) && state.opened) {
+      setState({
+        ...state,
+        opened: false,
+      });
+    }
+  };
```

Comment:
This code doesn't seem to work. I'm unable to close the dialog by clicking outside of the element. (Tested in both Chrome and Edge)

Reply:
Should work now :)